### ISSUE TITLE
[QUESTION] User not found error when calling `get_chat_member`

### ISSUE THREAD

**Issue by Strelkoveg** [python-telegram-bot/python-telegram-bot#3896]
### Issue I am facing

In some chats I get error when I call ```context.bot.get_chat_member(chat_id, user_id)```, even if user with given id is definetly exist in given chat. Sometimes it works fine, sometimes I get error. In some chats I don't have this error at all.

Why this happens? How to solve this problem?
Thanks!

### Traceback to the issue

```python
2023-09-22 08:34:18,836 - telegram.ext._application - ERROR - No error handlers are registered, logging exception.
Traceback (most recent call last):
  File "/home/strelkove/.local/lib/python3.10/site-packages/telegram/ext/_application.py", line 1115, in process_update
    await coroutine
  File "/home/strelkove/.local/lib/python3.10/site-packages/telegram/ext/_handler.py", line 141, in handle_update
    return await self.callback(update, context)
  File "/home/strelkove/pidor_bot/pidor.py", line 253, in pidor
    user_info = await context.bot.get_chat_member(chat_id, pidor_id)
  File "/home/strelkove/.local/lib/python3.10/site-packages/telegram/ext/_extbot.py", line 1565, in get_chat_member
    return await super().get_chat_member(
  File "/home/strelkove/.local/lib/python3.10/site-packages/telegram/_bot.py", line 368, in decorator
    result = await func(*args, **kwargs)  # skipcq: PYL-E1102
  File "/home/strelkove/.local/lib/python3.10/site-packages/telegram/_bot.py", line 3990, in get_chat_member
    result = await self._post(
  File "/home/strelkove/.local/lib/python3.10/site-packages/telegram/_bot.py", line 456, in _post
    return await self._do_post(
  File "/home/strelkove/.local/lib/python3.10/site-packages/telegram/ext/_extbot.py", line 312, in _do_post
    return await super()._do_post(
  File "/home/strelkove/.local/lib/python3.10/site-packages/telegram/_bot.py", line 487, in _do_post
    return await request.post(
  File "/home/strelkove/.local/lib/python3.10/site-packages/telegram/request/_baserequest.py", line 165, in post
    result = await self._request_wrapper(
  File "/home/strelkove/.local/lib/python3.10/site-packages/telegram/request/_baserequest.py", line 328, in _request_wrapper
    raise BadRequest(message)
telegram.error.BadRequest: User not found
```

### Related part of your code

```python
async def stats(update: Update, context: ContextTypes.DEFAULT_TYPE):
    chat_id = update.message.chat_id
    statistics = get_stats(update.message.chat_id, 'stats')
    if statistics == 'Ни один пользователь не зарегистрирован, статистики нет':
        await context.bot.send_message(chat_id=update.effective_chat.id, text=statistics)
    else:
        usernames = []
        counts = []
        for item in statistics.items():
            user_info = await context.bot.get_chat_member(chat_id, item[0])
            usernames.append(f'{user_info.user.full_name} (@{user_info.user.username})')
            counts.append(item[1])
        user_stats = dict(zip(usernames, counts))
        text_list = ['Результаты игры красавчик дня:']
        for j in dict(sorted(user_stats.items(), key=lambda sort_item: sort_item[1], reverse=True)).items():
            text_list.append(f'{j[0]}: {j[1]}')
        text = '\n'.join(text_list)
        await context.bot.send_message(chat_id=update.effective_chat.id, text=text)
```

### Operating System

Linux version 5.15.0-1021-aws (buildd@lcy02-amd64-045) (gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0, GNU ld (GNU Binutils for Ubuntu) 2.34) #25~20.04.1-Ubuntu SMP Thu Sep 22 13:59:08 UTC 2022

### Version of Python, python-telegram-bot & dependencies

```shell
python 3.10, python-telergam-bot 20.2
```

**Comment by Bibo-Joshi** [comment]
Hi. AFAIK the user must have started the bot for this approach to work. If you try to run `get_chat_member` for a user that has not started the bot, the bot can not find the user in a chat, even if it is a member of it.

**Comment by Strelkoveg** [comment]
Let me explain my case more clearly.
User adds chat bot ot chat, users run `reg` command. Here is the function
```async def reg(update: Update, context: ContextTypes.DEFAULT_TYPE):
    chat_id = update.message.chat_id
    reg_member = update.message.from_user.id
    user_info = await context.bot.get_chat_member(chat_id, reg_member)
    success_or_not = create_user(chat_id, reg_member)
    if success_or_not:
        message = f"{user_info.user.full_name}, ты в игре"
    else:
        message = f"{user_info.user.full_name}, зачем тебе регаться ещё раз?"
    await context.bot.send_message(chat_id=update.effective_chat.id, text=message)```
This command adds row on table 'stats' chat id of chat where command was executed and user id of user who executed the command.

When user leaves the chat, I delete corresponding row from table stats. Here is the function
```async def member_left(update: Update, context: ContextTypes.DEFAULT_TYPE):
    chat_id = update.message.chat_id
    reg_member = update.message.left_chat_member.id
    user_info = await context.bot.get_chat_member(chat_id, reg_member)
    message = unreg_in_data(chat_id, reg_member)
    if message == 'Пользователь не найден':
        await context.bot.send_message(chat_id=update.effective_chat.id, text='Мы не будем по нему скучать, '
                                                                              'ведь он не был в игре🤡')
    else:
        await context.bot.send_message(chat_id=update.effective_chat.id,
                                       text=f'{user_info.user.full_name} c позором бежал, но статистика всё помнит')```

So I always have all user_id who are actually in chat.
Then anyone in chat run function `stats` (it described on original question), which uses `get_chat_member` function with given chat_id and user_id from table. 
At this point I have a problem described above. I debugged it and see that I pass right arguments to function `get_chat_member` even if it fails. More over, it may fail on one user_id or another, and sometimes it works fine. In some chats it works fine everytime without errors.

**Comment by Bibo-Joshi** [comment]
How do you track a user leaving the chat? Do you also track the my_chat_member updates to see if a user blocks your bot? My suspicion is that some user IDs in your database are no longer accessible to your bot at some point.

**Comment by Strelkoveg** [comment]
I use `update.message.left_chat_member` to track user leaving from chat. I don't track users who block bot. But I checked bot's behavoiur when user blocks the bot, and I got another error, which clearly says that bot was bocked
```
2023-09-25 06:00:03,030 - telegram.ext._application - ERROR - No error handlers are registered, logging exception.
Traceback (most recent call last):
  File "/home/strelkove/.local/lib/python3.10/site-packages/telegram/ext/_application.py", line 1115, in process_update
    await coroutine
  File "/home/strelkove/.local/lib/python3.10/site-packages/telegram/ext/_handler.py", line 141, in handle_update
    return await self.callback(update, context)
  File "/home/strelkove/pidor_bot/pidor.py", line 249, in pidor
    await context.bot.send_message(chat_id=update.effective_chat.id, text='Невозможно выбрать пидора, список '
  File "/home/strelkove/.local/lib/python3.10/site-packages/telegram/ext/_extbot.py", line 2618, in send_message
    return await super().send_message(
  File "/home/strelkove/.local/lib/python3.10/site-packages/telegram/_bot.py", line 368, in decorator
    result = await func(*args, **kwargs)  # skipcq: PYL-E1102
  File "/home/strelkove/.local/lib/python3.10/site-packages/telegram/_bot.py", line 797, in send_message
    return await self._send_message(
  File "/home/strelkove/.local/lib/python3.10/site-packages/telegram/ext/_extbot.py", line 494, in _send_message
    result = await super()._send_message(
  File "/home/strelkove/.local/lib/python3.10/site-packages/telegram/_bot.py", line 549, in _send_message
    result = await self._post(
  File "/home/strelkove/.local/lib/python3.10/site-packages/telegram/_bot.py", line 456, in _post
    return await self._do_post(
  File "/home/strelkove/.local/lib/python3.10/site-packages/telegram/ext/_extbot.py", line 312, in _do_post
    return await super()._do_post(
  File "/home/strelkove/.local/lib/python3.10/site-packages/telegram/_bot.py", line 487, in _do_post
    return await request.post(
  File "/home/strelkove/.local/lib/python3.10/site-packages/telegram/request/_baserequest.py", line 165, in post
    result = await self._request_wrapper(
  File "/home/strelkove/.local/lib/python3.10/site-packages/telegram/request/_baserequest.py", line 319, in _request_wrapper
    raise Forbidden(message)
telegram.error.Forbidden: Forbidden: bot was blocked by the user
```

**Comment by Strelkoveg** [comment]
1. I checked case when user blocks the bot. Bot still can retrieve user info from user who blocked the bot when function `get_chat_member(chat_id=some_group_id, user_id=user_who_blocked_bot_id)` is called.
2. Thanks, I'll do that later. For now, there are 3-5 users in different chats
3. Thanks, I'll keep that in mind, too
For now I handled exception and return only user id instead of name when I catch BadRequest.

More over, we figured one thing: if every user in chat run some command, everything warks fine. Then, some time later, exception raises again.
